### ISSUE TITLE
How to detect coverage network in test?

### ISSUE THREAD

**Issue by jklepatch** [sc-forks/solidity-coverage#427]
Some tests fail inexplicably on with solidity-coverage (yes I already read [this](https://github.com/sc-forks/solidity-coverage/blob/master/docs/faq.md#continuous-integration-installing-metacoin-on-travisci-with-coveralls)).

I would like to disable these tests for solidity-coverage with a if statement:

```
const id = await web3.eth.net.getId();
//Check `truffle-config.js` and grab network name from the `id`

if(network !== 'coverage') {
  //run tests that pose problem only with solidity-coverage
}
```

Problem is, the [default network configuration](https://github.com/sc-forks/solidity-coverage#network-configuration) does not indicate the networkId used by solidity-coverage...

If someone knows this networkId, or another method to detect that we are running solidit-coverage network in tests, I would like to know :)

**Comment by cgewecke** [comment]
@jklepatch Are you using 6.x or 7.x-beta? In the latter it's possible to set the network id explicitly in .solcover.js 's providerOptions.

For 6.x, an example of how to do this could be found previously at the ganache launch scripts at [aragon-court, here](https://github.com/sc-forks/aragon-court/blob/ba911340323b69a5f09e156d290a92895f7a470f/scripts/ganache-cli.sh).

**Comment by jklepatch** [comment]
Thanks, I am using 0.6x, I will wait for the stable 0.7 release.

I fixed my problem by replacing `contract.readFunction(args)` to `contract.readFunction.call(args)`. (Truffle Contract).

When not using the `call()` notation, this gets executed as a transaction instead of a call, and I dont get the return value of the function.

**Comment by cgewecke** [comment]
@jklepatch Ok sounds good. 

> When not using the call() notation, this gets executed as a transaction instead of a call, and I dont get the return value of the function.n

The source of that problem is fixed in 0.7.x-beta because we're no longer modifying any solidity function signatures or instrumenting dependencies.
